crowbar_framework/app/models/api/upgrade.rb

```diff
@@ -407,24 +410,24 @@ def upgrade_drbd_cluster(cluster)
         # First node to upgrade is DRBD slave. There might be more resources using DRBD backend
         # but the Master/Slave distribution might be different among them.
         # Therefore, we decide only by looking at the first DRBD resource we find in the cluster.
-        drbd_slave = ""
-        drbd_master = ""
+        drbd_slave = nil
+        drbd_master = nil
 
         drbd_nodes.each do |drbd_node|
           cmd = "LANG=C crm resource status ms-drbd-{postgresql,rabbitmq}\
           | sort | head -n 2 | grep \\$(hostname) | grep -q Master"
           out = drbd_node.run_ssh_cmd(cmd)
           if out[:exit_code].zero?
-            drbd_master = drbd_node.name
+            drbd_master = drbd_node
           else
-            drbd_slave = drbd_node.name
+            drbd_slave = drbd_node
           end
         end
-        return false if drbd_slave.empty?
+        return false if drbd_slave.nil?
 
         return false unless upgrade_first_cluster_node drbd_slave, drbd_master
 
-        return false unless upgrade_next_cluster_node drbd_master
+        return false unless upgrade_next_cluster_node drbd_master, drbd_slave
```

Comment:
inside `upgrade_next_cluster_node` we [already call](https://github.com/crowbar/crowbar-core/pull/935/files#diff-33ff816d38c460e2a65a4918bf6f94d0R377) `upgrade_first_cluster_node` (line above).